src/base/zonefile_fmt.rs

```diff
@@ -11,21 +11,32 @@ impl From<fmt::Error> for Error {
 
 pub type Result = core::result::Result<(), Error>;
 
+pub enum DisplayKind {
+    Simple,
+    Aligned,
+    Multiline,
+}
+
 pub struct ZoneFileDisplay<'a, T: ?Sized> {
     inner: &'a T,
-    pretty: bool,
+    kind: DisplayKind,
 }
 
 impl<T: ZonefileFmt + ?Sized> fmt::Display for ZoneFileDisplay<'_, T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if self.pretty {
-            self.inner
-                .fmt(&mut MultiLineWriter::new(f))
-                .map_err(|_| fmt::Error)
-        } else {
-            self.inner
+        match self.kind {
+            DisplayKind::Simple => self
+                .inner
                 .fmt(&mut SimpleWriter::new(f))
-                .map_err(|_| fmt::Error)
+                .map_err(|_| fmt::Error),
+            DisplayKind::Aligned => self
+                .inner
+                .fmt(&mut AlignedWriter::new(f))
+                .map_err(|_| fmt::Error),
+            DisplayKind::Multiline => self
+                .inner
+                .fmt(&mut MultiLineWriter::new(f))
+                .map_err(|_| fmt::Error),
```

Comment:
Maybe the match arms could just invoke the appropriate `new()` and then after the match do `self.inner.fmt(&mut new_writer).map_err(|_| fmt::Error)`?

Reply:
I don't think that works without `dyn` unfortunately.